src/modules/emote_autocomplete/hooks/ChatInput.jsx

```diff
@@ -0,0 +1,48 @@
+import {useEffect, useState} from 'react';
+import emoteMenuViewStore from '../../../common/stores/emote-menu-view-store.js';
+
+function handleChatInput(focusedWord) {
+  if (focusedWord == null) {
+    return [];
+  }
+
+  const strippedFocusedWord = focusedWord.replace(/\s|:/g, '');
+  const foundEmotes = emoteMenuViewStore.search(strippedFocusedWord);
+
+  return foundEmotes.map((emote) => emote.item);
+}
+
+export default function useChatInput(chatInputElement, getAutocomplete) {
```

Comment:
this hook name doesn't make much sense. `useChatInput` implies you are retrieving the chat input, but that is not what is happening here. I would recommend refactoring this logic:

1. rename to `useChatInputPartialEmote`
2. make this hook responsible solely for propagating and returning input changes corresponding to partial emotes in the chat.
3. move store listening logic to another hook, if need be. more ideally, consider making a hook for listening to a store and unlistening from it on unmount and using it where we need updates to the store.

usage could look like:
```
const partialEmoteCode = useChatInputPartialEmote(chatInputElement, getChatInputPartialEmoteCode);
const emotes = useEmoteMenuViewStore((emotes) => emotes.search(partialEmoteCode).map((emote) => emote.item))
```